index.js

```diff
@@ -73,11 +93,26 @@ class TemplateCompiler extends Filter {
   }
 
   processString(string, relativePath) {
+    let srcDir = this.inputPaths[0];
+    let srcName = path.join(srcDir, relativePath);
     try {
-      return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
+      let result = 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
         contents: string,
-        moduleName: relativePath
+        moduleName: relativePath,
+        parseOptions: {
```

Comment:
`parseOptions` is only passed through with very recent `@glimmer/syntax` versions (it was added in https://github.com/glimmerjs/glimmer-vm/pull/871), this will have no effect in Ember versions less than 3.10.0.

I'd suggest using `env.meta.moduleName` instead (which has been supported for quite a while).

Reply:
Yes, but I still think this should be populated so the AST is properly configured.

Reply:
Maybe 🤔. We have avoided populating `srcName` because the semantics aren't really correct from a dev perspective. We don't have access to the original on disk location (which is what its supposed to be representing).

Reply:
> We don't have access to the original on disk location (which is what its supposed to be representing).

My take is that this information exists so that good error messages can be provided by ast transforms. All my existing error reporting code is based on taking in an AST node and using this source location information present there. If we take leave this out, I have to wire in an alternate path to get the source file information into the error reporter.

My more meta-level comment on this is that I think that the stack traces and paths that come out of errors  raised during broccoli addons should return the file name within the tree they were working on and that some sort of error display/logging/post-processing should translate that back into the filenames the user uses. broccoli trees that emit files with sourcemaps would make this approach solvable in a generic way.

Reply:
> My more meta-level comment on this is that I think that the stack traces and paths that come out of errors raised during broccoli addons should return the file name within the tree they were working on and that some sort of error display/logging/post-processing should translate that back into the filenames the user uses. broccoli trees that emit files with sourcemaps would make this approach solvable in a generic way.

Yes, I totally agree with this. 

From the perspective of this PR I think passing `parserOptions` is fine, but it will have no effect unless the app is using Ember 3.10+. I'm still unsure how you plan to deal with the missing `srcName` in the AST plugins you are authoring, but thats obviously a separable concern.